field/src/field_types.rs

```diff
@@ -91,6 +92,13 @@ pub trait Field:
         *self * (Self::ONE + Self::TWO)
     }
 
+    fn elementwise_product(lhs: &[Self], rhs: &[Self]) -> Vec<Self> {
```

Comment:
We already have `field::batch_util::batch_multiply_inplace`, which internally uses vectorization where possible.

Reply:
Oh good point, done